service/api/service.go

```diff
@@ -134,18 +152,37 @@ func (s *serviceImpl) ApiV1WorkflowStartPost(
 		InitSearchAttributes:               initCustomSAs,
 		InitDataAttributes:                 initCustomDAs,
 		Config:                             workflowConfig,
-		UseMemoForDataAttributes:           useMemo,
+		UseMemoForDataAttributes:           useMemoForDAs,
 		WaitForCompletionStateExecutionIds: req.GetWaitForCompletionStateExecutionIds(),
 		WaitForCompletionStateIds:          req.GetWaitForCompletionStateIds(),
 		OmitVersionMarker:                  s.config.Api.OptimizedVersioning,
 	}
 
 	runId, err := s.client.StartInterpreterWorkflow(ctx, workflowOptions, input)
 	if err != nil {
-		return nil, s.handleError(err, WorkflowStartApiPath, req.GetWorkflowId())
-	}
+		shouldHandleError := true
 
-	s.logger.Info("Started workflow", tag.WorkflowID(req.WorkflowId), tag.WorkflowRunID(runId))
+		if s.client.IsWorkflowAlreadyStartedError(err) && ignoreAlreadyStartedError {
+			response, descErr := s.client.DescribeWorkflowExecution(ctx, req.GetWorkflowId(), "", nil)
```

Comment:
I think we can check `requestId` before calling this API. If `requestId` is not provided, we can simply return 200 directly.

Reply:
Addressed